jaclang_jaseci/collections/base.py

```diff
@@ -0,0 +1,347 @@
+"""BaseCollection Interface."""
+
+from os import getenv
+from typing import (
+    Any,
+    AsyncGenerator,
+    Awaitable,
+    Generic,
+    Iterable,
+    Mapping,
+    Optional,
+    TypeVar,
+    Union,
+)
+
+from bson import ObjectId
+
+from motor.motor_asyncio import (
+    AsyncIOMotorClient,
+    AsyncIOMotorClientSession,
+    AsyncIOMotorCollection,
+    AsyncIOMotorCursor,
+    AsyncIOMotorDatabase,
+)
+
+from pymongo import (
+    DeleteMany,
+    DeleteOne,
+    IndexModel,
+    InsertOne,
+    UpdateMany,
+    UpdateOne,
+)
+from pymongo.server_api import ServerApi
+
+from ..utils import logger
+
+T = TypeVar("T")
+
+
+class BaseCollection(Generic[T]):
```

Comment:
Should we create a parent interface class for this that is agnostic to the specific database vendor? And have a mongodb one inherit from it?

Reply:
Maybe we can do this later...something to consider.